pkg/cmd/experimental/diagnostics/diagnostics.go

```diff
@@ -116,45 +118,48 @@ func NewCommandDiagnostics(name string, fullName string, out io.Writer) *cobra.C
 	cmd.Flags().BoolVar(&o.IsHost, options.FlagIsHostName, false, "look for systemd and journald units even without master/node config")
 	flagtypes.GLog(cmd.Flags())
 	options.BindLoggerOptionFlags(cmd.Flags(), o.LogOptions, options.RecommendedLoggerOptionFlags())
-	options.BindDiagnosticFlag(cmd.Flags(), &o.RequestedDiagnostics, options.NewRecommendedDiagnosticFlag())
 
 	return cmd
 }
 
 // Complete fills in DiagnosticsOptions needed if the command is actually invoked.
-func (o *DiagnosticsOptions) Complete() error {
+func (o *DiagnosticsOptions) Complete(args []string) error {
 	var err error
 	o.Logger, err = o.LogOptions.NewLogger()
 	if err != nil {
 		return err
 	}
 
-	return nil
-}
+	// If not given master/client config file locations, check if the defaults exist
+	// and adjust the options accordingly:
+	if len(o.MasterConfigLocation) == 0 {
+		if _, err := os.Stat(StandardMasterConfigPath); !os.IsNotExist(err) {
+			o.MasterConfigLocation = StandardMasterConfigPath
```

Comment:
If someone's got a perfectly fine master config file at a non-default position, forgets to add it as a flag, and has a broken file in the default position, will the diagnostic tool show the path to the file it's checking when it shows errors for that file? We should either do that or announce here that we've found a default and are going to use that.